File: RPC/RPC_10_12_10_2024/Soluciones/ThirteenthAnnual_UCF_LocalProgrammingContest/local2024FinalRoundData/solutions/G_speed/speed.py

n,L = map(int, input().split())

boosts = [list(int(x) for x in input().split()) for _ in range(n)]


boosts.sort()

timeToGet = [a[0] for a in boosts]

ans = L
for i in range(n):
   for j in range(i+1,n):
      if boosts[i][0] + boosts[i][2]*boosts[i][1] <= boosts[j][0]:
         timeToGet[j] = min(timeToGet[j], timeToGet[i] + boosts[i][2] + (boosts[j][0] - boosts[i][0] - boosts[i][1]*boosts[i][2]))
   if boosts[i][0] + boosts[i][2]* boosts[i][1] <= L:
      ans = min(ans, timeToGet[i] + (L - boosts[i][0] - boosts[i][1]*boosts[i][2]) + boosts[i][2])
   else:
      ans = min(ans, timeToGet[i] + (L - boosts[i][0]) / boosts[i][1])
print(ans)

File: RPC/RPC_10_12_10_2024/Soluciones/ThirteenthAnnual_UCF_LocalProgrammingContest/local2024FinalRoundData/solutions/G_speed/speed.cpp

#include <bits/stdc++.h>

using namespace std;

using ll = long long;
using ld = long double;
using vi = vector<int>;
using vvi = vector<vi>;
using vd = vector<ld>;

int main() {
   int n;
   ll L;
   cin >> n >> L;
   vvi boosts(n, vi(3));
   for (int i = 0; i < n; i++) {
      for (int j = 0; j < 3; j++) {
         cin >> boosts[i][j];
      }
   }
   sort(boosts.begin(), boosts.end());

   vd times(n);
   for (int i = 0; i < n; i++) times[i] = boosts[i][0];
   ld ans = L;
   for (int i = 0; i < n; i++) {
      int end = boosts[i][0] + boosts[i][1] * boosts[i][2];
      for (int j = i + 1; j < n; j++) {
         if (end <= boosts[j][0]) {
            times[j] = min(times[j], times[i] + boosts[j][0] - end + boosts[i][2]);
         }
      }
      if (end <= L) {
         ans = min(ans, times[i] + L - end + boosts[i][2]);
      } else {
         ans = min(ans, times[i] + (L-boosts[i][0]) / ((ld)(boosts[i][1])));
      }
   }

   cout << setprecision(12) << fixed << ans << endl;

   return 0;
}
